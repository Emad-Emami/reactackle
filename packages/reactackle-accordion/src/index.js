import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
