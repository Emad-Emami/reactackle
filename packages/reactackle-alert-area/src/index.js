import AlertArea from './AlertArea';

export { AlertArea };
