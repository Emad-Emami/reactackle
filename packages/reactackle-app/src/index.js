import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
