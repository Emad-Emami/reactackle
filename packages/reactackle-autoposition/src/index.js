import AutoPosition, { recalculateAllAutoPosition } from './AutoPosition';

export { AutoPosition, recalculateAllAutoPosition };
