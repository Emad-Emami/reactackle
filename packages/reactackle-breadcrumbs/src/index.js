import Breadcrumbs from './Breadcrumbs';

export { Breadcrumbs };
export * from './BreadcrumbsItem';
