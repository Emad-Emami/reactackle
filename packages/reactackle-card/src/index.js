import Card from './Card';

export { Card };
export * from './CardActions';
export * from './CardAreaMain';
export * from './CardAreaSide';
export * from './CardContent';
export * from './CardHeaderPrimary';
export * from './CardHeaderSecondary';
export * from './CardMedia';
