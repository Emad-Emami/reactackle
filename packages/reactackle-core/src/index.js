// theme
// defaultTheme
// composeReactackleTheme
// defaultThemeBlueprintBroadcast
// registerDefaultComponentTheme ++
// extractThemeOrDefault ++

// styleHelpers - for combining local theme
// export everything

// styleFunctions
// export everything

// - resetStyles
// - globalStyles
// - heladerClasses

// base
// export nothing

// constants
// export everything

// mixins
// export everything

// import {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// } from './theme';

// export {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// };

export * from './theme';
export * from './mixins';
export * from './styleFunctions';
export * from './styleHelpers';
export * from './utils';
export * from './constants/constants';
export * from './hoc/withExternalProps';
export * from './globalStyles';
export * from './resetStyles';
export * from './Theme/Theme';
export * from './Theme/withTheme';
