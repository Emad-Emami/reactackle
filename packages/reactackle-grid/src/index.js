'use strict';

import { registerDefaultComponentTheme } from 'reactackle-core';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('grid', componentTheme);

export * from './Container/Container';
export * from './Column/Column';
export * from './FlexRegion/FlexRegion';
export * from './PaddingBox/PaddingBox';
export * from './Row/Row';
