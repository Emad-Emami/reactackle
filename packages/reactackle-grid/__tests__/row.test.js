'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { Row } from '../src';

jest.mock('react-dom');

describe('<Row/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Row />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if layoutDirection set horizontal', () => {
    const tree = renderer.create(
      <Row layoutDirection="horizontal" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if layoutDirection set vertical', () => {
    const tree = renderer.create(
      <Row layoutDirection="vertical" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if behavior set nest', () => {
    const tree = renderer.create(
      <Row behavior="nest" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if behavior set collapse', () => {
    const tree = renderer.create(
      <Row behavior="collapse" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if collapse set', () => {
    const collapse = {
        xsmall: true,
      },
      tree = renderer.create(
        <Row collapse={collapse} />,
      );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if expandRatio set 0.5', () => {
    const tree = renderer.create(
      <Row expandRatio={0.5} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if spacing set true', () => {
    const tree = renderer.create(
      <Row spacing />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
