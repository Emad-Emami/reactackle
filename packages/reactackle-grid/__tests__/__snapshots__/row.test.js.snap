// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row/> renders correctly if behavior set collapse 1`] = `
<div
  className="RowStyled-hOvpFe iQvgeE"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;

exports[`<Row/> renders correctly if behavior set nest 1`] = `
<div
  className="RowStyled-hOvpFe iQvgeE"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;

exports[`<Row/> renders correctly if collapse set 1`] = `
<div
  className="RowStyled-hOvpFe iQvgeE"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;

exports[`<Row/> renders correctly if expandRatio set 0.5 1`] = `
<div
  className="RowStyled-hOvpFe izfsux"
  style={
    Object {
      "flexGrow": 0.5,
    }
  }
/>
`;

exports[`<Row/> renders correctly if layoutDirection set horizontal 1`] = `
<div
  className="RowStyled-hOvpFe iQvgeE"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;

exports[`<Row/> renders correctly if layoutDirection set vertical 1`] = `
<div
  className="RowStyled-hOvpFe lmbZbe"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;

exports[`<Row/> renders correctly if spacing set true 1`] = `
<div
  className="RowStyled-hOvpFe iQvgeE"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;

exports[`<Row/> renders correctly with default props 1`] = `
<div
  className="RowStyled-hOvpFe iQvgeE"
  style={
    Object {
      "flexGrow": 0,
    }
  }
/>
`;
